operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/LeaderElectionConfiguration.java

```diff
@@ -59,8 +68,8 @@ public LeaderElectionConfiguration(
     this.identity = identity;
   }
 
-  public String getLeaseNamespace() {
-    return leaseNamespace;
+  public Optional<String> getLeaseNamespace() {
```

Comment:
Note: this is a breaking change. I can mark it as deprecated and provide another method `getOptionalLeaseNamespace`, if needed.

Reply:
Ahh normally I would prefer backwards compatible with deprecated. `getOptionalLeaseNamespace` does not sound too good, maybe just stick with this this time.

Reply:
Probably would be better, yes.

Reply:
I think for users, direct usage of this API will be rare even they do use leader election feature. Probably only in unit test.